crates/iota-transaction-checks/src/lib.rs

```diff
@@ -61,10 +61,37 @@ mod checked {
             gas,
             transaction.gas_budget(),
             transaction.gas_price(),
-            transaction.kind(),
+            Some(transaction.kind()),
```

Comment:
```suggestion
        if transaction.is_system_tx() {
            Ok(IotaGasStatus::new_unmetered())
        } else {
            check_gas(
                objects,
                protocol_config,
                reference_gas_price,
                gas,
                transaction.gas_budget(),
                transaction.gas_price(),
            )
```
let's move this check here, so that you don't need to pass tx_kind to `check_gas`

Reply:
Done